lib/ansible/inventory/group.py

```diff
@@ -72,16 +71,20 @@ def deserialize(self, data):
         self.name = data.get('name')
         self.vars = data.get('vars', dict())
         self.depth = data.get('depth', 0)
-        self.hosts = data.get('hosts', {})
-
-        self._hosts = set(self.hosts)
```

Comment:
`self._hosts` must be set to `None` here or in `__init__`, otherwise it will not be initialized by the `host_names` property after deserialization.